etl/steps/data/garden/climate/2024-01-31/ghg_concentration.py

```diff
@@ -0,0 +1,140 @@
+            .rolling(365)
+            .mean()
+            .copy_metadata(tb_with_average[original_column_name])
+        )
+
+    # Drop empty rows.
+    tb_with_average = tb_with_average.dropna(subset=original_column_names, how="all").reset_index()
+
+    # Remove rolling average for the first year, given that it is based on incomplete data.
+    for original_column_name in original_column_names:
+        tb_with_average.loc[
+            tb_with_average["date"] < tb_with_average["date"].min() + pd.Timedelta(days=365),
+            f"{original_column_name}_yearly_average",
+        ] = None
+
+    # Sort conveniently.
+    tb_with_average = tb_with_average.sort_values(["location", "date"]).reset_index(drop=True)
+
+    for original_column_name in original_column_names:
+        # Check that the values of the original column have not been altered.
+        error = f"The values of the original {original_column_name} column have been altered."
+        assert tb_with_average[original_column_name].astype(float).equals(tb[original_column_name].astype(float)), error
+
+    return tb_with_average
+
+
+def prepare_gas_data(tb: Table) -> Table:
+    tb = tb.copy()
+
+    # Extract gas name from table's short name.
+    gas = tb.metadata.short_name.split("_")[0]
+
+    # Columns to select from the data, and how to rename them.
+    columns = {
+        "year": "year",
+        "month": "month",
+        "average": f"{gas}_concentration",
+        # The following column is loaded only to perform a sanity check.
+        "decimal": "decimal",
+    }
+
+    # Select necessary columns and rename them.
+    tb = tb[list(columns)].rename(columns=columns, errors="raise")
+
+    # There is a "decimal" column for the year as a decimal number, that only has 12 possible values, corresponding to
+    # the middle of each month, so we will assume the 15th of each month.
+    error = "Date format has changed."
+    assert len(set(tb["decimal"].astype(str).str.split(".").str[1])) == 12, error
+    assert set(tb["month"]) == set(range(1, 13)), error
+    tb["date"] = pd.to_datetime(tb[["year", "month"]].assign(day=15))
+
+    # Remove unnecessary columns.
+    tb = tb.drop(columns=["year", "month", "decimal"], errors="raise")
+
+    # Add a location column.
+    tb["location"] = "World"
+
+    # Add a column with a rolling average for each gas.
+    tb = add_rolling_average(tb=tb, original_column_names=[f"{gas}_concentration"])
+
+    return tb
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset and read its main table.
+    ds_meadow = paths.load_dataset("ghg_concentration")
+    tb_co2 = ds_meadow["co2_concentration_monthly"].reset_index()
+    tb_ch4 = ds_meadow["ch4_concentration_monthly"].reset_index()
+    tb_n2o = ds_meadow["n2o_concentration_monthly"].reset_index()
+
+    #
+    # Process data.
+    #
+    # Prepare data for each gas.
+    tb_co2 = prepare_gas_data(tb=tb_co2)
```

Comment:
Yes, I manually imposed that in `add_rolling_average`, given that the first 12 data points do not have enough data to have a full year of a rolling average. Do you think the points should be there? I think it would create some spurious oscillations in the annual average on those first points.

Reply:
Ah I was just curious about the difference between this and current explorer, as the datapoints seem to be there currently. Were they calculated differently? Not a big deal either way, just wanted to check it was intentional.

Reply:
Yes, the yearly averages were calculated in a slightly different way. I'd need to go back to importers to check how this was calculated. But my intuition is that the old one is not exactly an annual rolling average. If you look at [these last 12 points of the N2O concentration](https://ourworldindata.org/explorers/climate-change?time=2022-10-15..latest&facet=none&Metric=N%E2%82%82O+concentrations&Long-run+series=false&country=OWID_WRL~ATA~Gulkana+Glacier~Lemon+Creek+Glacier~OWID_NAM~South+Cascade+Glacier~Wolverine+Glacier~Hawaii) in production, you see that the last point of the annual average is 337, which is larger than any of the last 12 points. I've also checked that the new yearly average is well calculated, namely that the last data point corresponds to the mean of the last 12 points:
![climate-change](https://github.com/owid/etl/assets/12246978/3a6da268-f74f-443b-8f28-259836d58288)